extensions/identity-hub-verifier/src/main/resources/META-INF/services/org.eclipse.dataspaceconnector.spi.system.ServiceExtension

```diff
@@ -0,0 +1,2 @@
+org.eclipse.dataspaceconnector.identityhub.verifier.CredentialsVerifierExtension
+org.eclipse.dataspaceconnector.identityhub.verifier.JwtCredentialsVerifierExtension
```

Comment:
is it required to split into 2 extensions?

Reply:
Yes, otherwise `CredentialsVerifierExtensionTest` fails with a NullPointerException, probably due to some mysterious incorrect DI object instantiation ordering. @paullatzelsperger would be good to have a look at this one together.

Reply:
For this PR, I will just document the reasoning of requiring 2 separate extensions. We can look at the issue separately.

Reply:
The problem is the following: if all `@Provider` methods are in one extension, then we have a situation where the same extension both provides and injects the same service (i.e. the `@JwtCredentialsVerifier`).
While resolvin the `@Inject`, the provider method gets called, at which point the (also injected) `DidPublicKeyResolver` is still `null`, because it's not yet resolved...

There are two options to solve this:
1. change the order of field declaration: simply declare the `DidPublicKeyResolver` _above_ the `JwtCredentialsVerifier`. **this is brittle and prone to errors! do not do this!**
2. inject the `DidPublicKeyResolver` "lazily" in the `@Provider` method:
   ```java
    @Provider(isDefault = true)
    public JwtCredentialsVerifier createJwtVerifier(ServiceExtensionContext context) {
        if(didPublicKeyResolver == null){
            didPublicKeyResolver = context.getService(DidPublicKeyResolver.class);
        }
        return new DidJwtCredentialsVerifier(didPublicKeyResolver, monitor);
    }
    ```
   note that `didPublicKeyResolver` is a normal field, without the `@Inject` annotation on it.